src/common-ui/components/ToggleSwitch.js

```diff
@@ -17,18 +17,35 @@ class ToggleSwitch extends PureComponent {
         activeClassName: localStyles.activeSwitch,
     }
 
+    constructor(props) {
+        super(props)
+
+        this.state = {
+            isChecked: props.isChecked,
+        }
+    }
+
     deriveClass = () =>
         cx(this.props.className, {
             [this.props.activeClassName]: this.props.isChecked,
         })
 
+    handleClick = event => {
+        this.setState(state => ({
```

Comment:
These changes to `ToggleSwitch` are not what we talked about last week. What it needs is **optional support** for being a uncontrolled component.
Here's how it could work, trying to be like how React's controlled and uncontrolled inputs work:
- allow `props.isChecked` to be optional; that prop affords outside control of the state if it's defined
- if it's undefined, you should attempt to control the state locally (like you're doing now in `handleClick` event handler but with a condition)
- add an optional `props.defaultValue` bool prop. This should be what your initial local state is set to, not `props.isChecked` (you should also specify `static defaultProps` to ensure it has a sane default in the case that the user doesn't specify it). You also shouldn't need to do this in the `constructor`
- update the `deriveClass` method to check `props.isChecked || state.isChecked` 

Once it works like that (assuming I haven't missed anything), you should then be able to remove your `shouldTrack` notifications UI state.